core/drivers/amd/mailbox_driver.c

```diff
@@ -0,0 +1,216 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2024-2025, Advanced Micro Devices, Inc. All rights reserved.
+ *
+ */
+#include <drivers/amd/mailbox_driver.h>
+#include <initcall.h>
+#include <kernel/delay.h>
+#include <kernel/panic.h>
+#include <mailbox_private.h>
+#include <malloc.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <string.h>
+#include <tee/cache.h>
+
+static struct ipi_info_t ipi_info;
+
+static TEE_Result mailbox_call(enum mailbox_api ipi_api, uint32_t blocking)
+{
+	struct thread_smc_args args = {
+		.a0 = PLAT_SMC_SIP_IPI | ipi_api,
+		.a1 = reg_pair_to_64(0, ipi_info.local),
+		.a2 = reg_pair_to_64(0, ipi_info.remote),
+		.a3 = reg_pair_to_64(0, blocking),
+	};
+
+	thread_smccc(&args);
+
+	if (args.a0) {
+		EMSG("Mailbox Call returned with error 0x%lx", args.a0);
```

Comment:
`args.a0` is always `uint64_t` in the case of this driver, if I'm right, so prefer `PRIx64`.
`		EMSG("Mailbox Call returned with error 0x%"PRIx64, args.a0);`

That said, if 32bit core is to be considered (I see that plat-versal2/conf.mk allows that: `CFG_ARM64_core ?= y`) then maybe prefer using a cast to ensure both cases are supported:
`		EMSG("Mailbox Call returned with error 0x%lx", (unsigned long)args.a0);`